src/main/java/com/code_intelligence/jazzer/mutation/mutator/collection/MapMutatorFactory.java

```diff
@@ -126,32 +130,101 @@ public void initInPlace(Map<K, V> reference, PseudoRandom prng) {
       }
     }
 
+    private void eraseRandomChunk(Map<K, V> map, PseudoRandom prng, int mapSize) {
+      int upperLimit = Math.max(1, Math.min(mapSize - minSize, mapSize / 2));
+      int chunkSize = upperLimit == 1 ? 1 : prng.closedRange(1, upperLimit);
+      int chunkOffset = prng.closedRange(0, mapSize - chunkSize);
+      List<K> tmpRemStore = new ArrayList<>(chunkSize);
+      map.entrySet().stream().skip(chunkOffset).limit(chunkSize).forEach(entry -> {
```

Comment:
Map iteration order will in general be non-deterministic. We could force the maps we generated to be `LinkedHashMap`s, which would make the mutations deterministic. 

@bertschneider What do you think?

Reply:
True, `LinkedHashMap` would help with ordering and shouldn't introduce too much of a performance penalty. I think it's a good idea to switch.

Reply:
@fmeum @bertschneider Is that something to be done in this PR or is that "meta" enough to extract this into a separate one?

Reply:
I can do it as a follow-up.